compiler/rustc_errors/src/emitter.rs

```diff
@@ -1383,7 +1384,27 @@ impl HumanEmitter {
                 buffer.append(0, level.to_str(), Style::MainHeaderMsg);
                 buffer.append(0, ": ", Style::NoStyle);
             }
-            self.msgs_to_buffer(&mut buffer, msgs, args, max_line_num_len, "note", None);
+            let printed_lines = self.msgs_to_buffer(&mut buffer, msgs, args, max_line_num_len, "note", None);
```

Comment:
The commit that introduces the `|` to multline `note`s doesn't need to land with this PR at all. I can even leave the ASCII art as is and only put the vertical line in the unicode output exclusively.

Reply:
Reverted in a subsequent commit.